src/systems/swarm_behavior.rs

```diff
@@ -0,0 +1,179 @@
+use amethyst::{
+    core::{
+        nalgebra::Vector3,
+        timing::Time,
+        transform::components::{Parent, Transform},
+    },
+    ecs::*,
+    shrev::EventChannel,
+};
+
+use rand::{thread_rng, Rng};
+use std::f32;
+
+use crate::{
+    components::{
+        creatures::{Movement, Wander},
+        swarm::{SwarmBehavior, SwarmCenter},
+    },
+    systems::spawner::CreatureSpawnEvent,
+};
+
+#[derive(Default)]
+pub struct SwarmSpawnSystem {
+    swarm_timer: f32,
+}
+
+impl<'s> System<'s> for SwarmSpawnSystem {
+    type SystemData = (
+        Entities<'s>,
+        Read<'s, LazyUpdate>,
+        Write<'s, EventChannel<CreatureSpawnEvent>>,
+        Read<'s, Time>,
+    );
+
+    fn run(&mut self, (entities, lazy_update, mut spawn_events, time): Self::SystemData) {
+        let delta_seconds = time.delta_seconds();
+        self.swarm_timer -= delta_seconds;
+        if self.swarm_timer <= 0.0 {
+            let mut swarm_entity_builder = lazy_update.create_entity(&entities);
+            let mut rng = thread_rng();
```

Comment:
Not really a review concerning the code, but sounds like a good opportunity to raise this point: we have to be mindful about RNG. 

One good feature to have is determinism, this allows us to repeat the same simulation with the same results, it makes testing easier, a player can easily share his simulation by sharing the initial seed (like Minecraft) and tweaking becomes more fun because players can run the same simulation but tweak a few things and see how it changes.

From the `thread_rng()` docs:
```
After generating a certain amount of randomness, the RNG will reseed itself from the operating system or, if the operating system RNG returns an error, a seed based on the current system time.

The internal RNG used is platform and architecture dependent, even if the operating system random number generator is rigged to give the same sequence always. If absolute consistency is required, explicitly select an RNG, e.g. IsaacRng or Isaac64Rng.
```

It seems that `thread_rng()` isn't deterministic at the very least between different platforms.

Of course that's open to discussion, do we really want this Feature? And there is no reason to change until we decide this and roll a deterministic RNG that suits our use case.

Reply:
+1 for using a resource for rng.
+2 for using an rng component with different seeds for unrelated stuff, so that the systems will not wait for each other (rng would need to be `Write`)
@khskarl This might be an interesting topic to discuss in the forums so that it gets more visibility.

Reply:
Yeah this is an important design discussion as well so please move it to the forum.

Reply:
Design discussion continued here:

https://community.amethyst.rs/t/determinism-and-being-mindful-of-rng/826?u=erlend_sh